clang/lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -750,10 +750,11 @@ NVPTXToolChain::TranslateArgs(const llvm::opt::DerivedArgList &Args,
     if (!llvm::is_contained(*DAL, A))
       DAL->append(A);
 
-  // TODO: We should accept 'generic' as a valid architecture.
   if (!DAL->hasArg(options::OPT_march_EQ) && OffloadKind != Action::OFK_None) {
     DAL->AddJoinedArg(nullptr, Opts.getOption(options::OPT_march_EQ),
                       CudaArchToString(CudaArch::CudaDefault));
+  } else if (DAL->getLastArgValue(options::OPT_march_EQ) == "generic") {
```

Comment:
```suggestion
  } else if (DAL->getLastArgValue(options::OPT_march_EQ) == "generic"
                 && OffloadKind == Action::OFK_None) {
```
Ah, forgot, we probably don't want to expose this to CUDA just yet.